### DISCUSSION TITLE
Please help me with this code:

### DISCUSSION THREAD

**Discussion by mango420** [Rapptz/discord.py#6526] [Q&A]
Hey Guys!

I don't know why, but a part of my code stoped working. Yesterday it worked all fine but today it doesn't work anymore. I wan't the bot to do: Every time, a user sends "b.avatar @user" the bot respondes with the avatar of the mentioned user. 

code:
import discord, sys, os, time, json, requests

client = discord.Client()
admin = (604793540395925536)

@client.event
async def on_ready():
    print('Logged in as {0.user}'.format(client))
    await client.change_presence(activity=discord.Activity(type=discord.ActivityType.listening, name="Function Testscript / *******."))

@client.event
async def on_message(message):
    if message.author == client.user:
        return

    if message.content == ('b.avatar'):
        if (message.mentions.__len__()>0):
                for user in message.mentions:
                    await message.channel.send('Hier: ')
                    await message.channel.send(user.avatar_url)
        else:
            await message.channel.send('U have to mention someone')

client.run('my_token')

again it just stoped working and there is no error.

**Comment by mango420** [comment]
Havet the answer.

**Comment by 15696** [comment]
The issue is with:
```py
if message.content == ('b.avatar'):
```

That means, if there is something after `b.avatar`, the message will be ignored. You should instead uses `if message.content.startswith("b.avatar")`, or even better, rewrite in the commands extension, which makes writing commands much easier since you don't have to parse them yourself. 

For example, your entire avatar command could be rewritten to: (this is not the full code, just the avatar command)
```py
@client.command()
async def avatar(ctx, member: discord.Member = None):
    if not member:
        return await ctx.send("You have to mention someone.")
    await ctx.send(f"Here: {member.avatar_url}")
```

Also, I see you have `time` and `requests` in your import. Depending on what you're using them for, they might be blocking, which [you should avoid](https://discordpy.readthedocs.io/en/latest/faq.html#what-does-blocking-mean).

**Reply by mango420** [reply]
thanks u helped me a looot. And can suggest me some websites were i can learn these "commands extensions". I watched some videos about Discord.py and they all say i should do it like this.

**Reply by bandwiches** [reply]
[Here You Go](https://discordpy.readthedocs.io/en/latest/ext/commands/index.html)

**Reply by 15696** [reply]
@CaptainMango1 Don't watch outdated videos. All videos on discord.py are mostly outdated.
